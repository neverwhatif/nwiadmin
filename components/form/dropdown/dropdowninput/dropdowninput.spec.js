import React from 'react';
import { shallow } from 'enzyme';

import DropdownInput from './index';

jest.mock('nwiadmin/utility');

describe('DropdownInput', () => {
    const component = shallow(<DropdownInput text="Lorem" />);
    const componentIsLoading = shallow(<DropdownInput text="Lorem" isLoading />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
        expect(componentIsLoading).toMatchSnapshot();
    });
    it('should render text', () => {
        expect(component.text()).toBe('Lorem');
    });
    it('should render loading', () => {
        expect(componentIsLoading.text()).toBe('Loading...');
    });
    it('should have a default onClick', () => {
        expect(DropdownInput.defaultProps.onClick()).toBeNull();
    });
    it('should handle onClick', () => {
        const mockOnClick = jest.fn();
        const componentWithClick = shallow(<DropdownInput text="Lorem" onClick={() => mockOnClick()} />);

        componentWithClick.simulate('click');
        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
    it('should handle onKeyDown', () => {
        const mockOnClick = jest.fn();
        const componentWithClick = shallow(<DropdownInput text="Lorem" onClick={() => mockOnClick()} />);

        componentWithClick.simulate('keyDown');
        expect(mockOnClick).toHaveBeenCalledTimes(1);
    });
});
