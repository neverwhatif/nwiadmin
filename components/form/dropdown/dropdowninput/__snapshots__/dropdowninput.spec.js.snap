// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownInput should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropdownInput
    isLoading={false}
    onClick={[Function]}
    text="Lorem"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="root"
        >
          Lorem
        </div>,
        <i
          className="control"
        />,
      ],
      "className": "root",
      "onClick": [Function],
      "onKeyDown": [Function],
      "role": "button",
      "tabIndex": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Lorem",
          "className": "root",
        },
        "ref": null,
        "rendered": "Lorem",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "control",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="root"
          >
            Lorem
          </div>,
          <i
            className="control"
          />,
        ],
        "className": "root",
        "onClick": [Function],
        "onKeyDown": [Function],
        "role": "button",
        "tabIndex": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Lorem",
            "className": "root",
          },
          "ref": null,
          "rendered": "Lorem",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "control",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`DropdownInput should render 2`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DropdownInput
    isLoading={true}
    onClick={[Function]}
    text="Lorem"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="root"
        >
          Loading...
        </div>,
        <i
          className="control"
        />,
      ],
      "className": "root",
      "onClick": [Function],
      "onKeyDown": [Function],
      "role": "button",
      "tabIndex": 0,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
          "className": "root",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "control",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="root"
          >
            Loading...
          </div>,
          <i
            className="control"
          />,
        ],
        "className": "root",
        "onClick": [Function],
        "onKeyDown": [Function],
        "role": "button",
        "tabIndex": 0,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
            "className": "root",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "control",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
